frontend/plan/components/search/SchoolReq.tsx

```diff
@@ -1,24 +1,43 @@
 import React, { useState } from "react";
 import PropTypes from "prop-types";
+import { FilterData, School, Requirement } from "../../types";
+
+interface SchoolReqProps {
+    startSearch: (searchObj: object) => void;
+    filterData: FilterData;
+    schoolReq: { [K in School]: Requirement[] };
+    addSchoolReq: (s: string) => void;
+    remSchoolReq: (s: string) => void;
+}
+
+type SchoolDisplay = "College" | "Engineering" | "Nursing" | "Wharton";
 
 export function SchoolReq({
     startSearch,
     filterData,
     schoolReq,
     addSchoolReq,
     remSchoolReq,
-}) {
-    const schools = ["College", "Engineering", "Nursing", "Wharton"];
-    const [selSchool, setSelSchool] = useState("College");
+}: SchoolReqProps) {
+    const schools: SchoolDisplay[] = [
+        "College",
+        "Engineering",
+        "Nursing",
+        "Wharton",
+    ];
+    const [selSchool, setSelSchool] = useState<SchoolDisplay>("College");
 
-    const schoolCode = new Map();
-    schoolCode.set("College", "SAS");
-    schoolCode.set("Engineering", "SEAS");
-    schoolCode.set("Wharton", "WH");
-    schoolCode.set("Nursing", "NURS");
+    const schoolCode: { [K in SchoolDisplay]: School } = {
+        College: School.COLLEGE,
```

Comment:
@eyingxuan can correct me if I'm wrong, but I believe that if you give enums a string value, you can just cast the string to the enum type and things will work as expected.

Reply:
I think the issue here was that the backend requires using an abbreviation of each school in the route (such as "SAS", "SEAS" etc) so School.COLLEGE is set to be equal to the abbreviated string, while we need what's displayed to be an actual string. 

What could be cleaner in retrospect is actually to write a function that turns the School enum into a display string, and use the enum itself in the useState to change between school requirements. What do you think? @davish @haoyxc

Reply:
i'll defer to you guys since you probably know the codebase better than I do :-)